heron/common/src/cpp/metrics/metricsmgr-client.cpp

```diff
@@ -63,6 +63,12 @@ void MetricsMgrClient::HandleClose(NetworkErrorCode) {
 
 void MetricsMgrClient::ReConnect() { Start(); }
 
+void MetricsMgrClient::SetPublisherPort(const sp_int32 _port) {
+  if (0 == port_ && 0 != _port) {
```

Comment:
@srkukarni I found that it needs alot refactoring to create clients after the server gets actual port. too many objects need metrics_manager_client_ and tmaster_client_ before StMgrServer binds to port 0.